src/main/java/com/iexec/worker/feign/LoginService.java

```diff
@@ -61,19 +61,20 @@ public String login() {
         String workerAddress = credentialsService.getCredentials().getAddress();
         ECKeyPair ecKeyPair = credentialsService.getCredentials().getEcKeyPair();
 
-        String challenge = "";
+        String challenge;
         try {
             challenge = coreClient.getChallenge(workerAddress);
         } catch (FeignException e) {
             log.error("Cannot login, failed to get challenge [status:{}]", e.status());
+            return "";
         }
         if (StringUtils.isEmpty(challenge)) {
             log.error("Cannot login, challenge is empty [challenge:{}]", challenge);
             return "";
         }
 
         Signature signature = SignatureUtils.hashAndSign(challenge, workerAddress, ecKeyPair);
-        String token = "";
+        String token;
```

Comment:
```suggestion
        final String token;
```
?

Reply:
impossible, this variable is updated in the try block